pkg/controller/perconaservermongodb/mgo.go

```diff
@@ -318,11 +321,29 @@ func (r *ReconcilePerconaServerMongoDB) handleReplsetInit(m *api.PerconaServerMo
 			return fmt.Errorf("get host for the pod %s: %v", pod.Name, err)
 		}
 
-		cmd := []string{
+		var errb, outb bytes.Buffer
+		cmd := []string{"sh", "-c", "mongod --version | head -1 | awk '{print $3}' | awk -F'.' '{print $1\".\"$2}' | cut -c2-"}
+		err = r.clientcmd.Exec(&pod, "mongod", cmd, nil, &outb, &errb, false)
+		if err != nil {
+			return fmt.Errorf("exec mongod --version: %v / %s / %s", err, outb.String(), errb.String())
+		}
+		mongoVer := v.Must(v.NewVersion(strings.TrimSpace(outb.String())))
+
+		mongoCmd := "mongo"
+		if !m.Spec.UnsafeConf {
+			// --tls* options is introduced in 4.2, we need to use --ssl* for older versions
+			if mongoVer.Compare(sslDeprecatedVersion) < 0 {
+				mongoCmd += " --ssl --sslPEMKeyFile /tmp/tls.pem --sslAllowInvalidCertificates --sslCAFile /etc/mongodb-ssl/ca.crt"
+			} else {
+				mongoCmd += " --tls --tlsCertificateKeyFile /tmp/tls.pem --tlsAllowInvalidCertificates --tlsCAFile /etc/mongodb-ssl/ca.crt"
+			}
```

Comment:
but these transformations are done inside the entrypoint. see - https://github.com/percona/percona-server-mongodb-operator/blob/main/build/ps-entry.sh#L420

I am not sure if we should duplicate logic here?
```suggestion
			mongoCmd += " --ssl --sslPEMKeyFile /tmp/tls.pem --sslAllowInvalidCertificates --sslCAFile /etc/mongodb-ssl/ca.crt"
```